packages/strapi-admin/admin/src/components/HeadersInput/index.js

```diff
@@ -0,0 +1,174 @@
+import CreatableSelect from 'react-select/creatable';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { CircleButton } from 'strapi-helper-plugin';
+import { InputText } from '@buffetjs/core';
+import { Plus } from '@buffetjs/icons';
+
+import keys from './keys';
+import Wrapper from './Wrapper';
+
+const HeadersInput = ({
+  errors,
+  name,
+  onBlur,
+  onClick,
+  onChange,
+  onRemove,
+  value,
+}) => {
+  const optionFormat = value => ({ value: value, label: value });
+  const options = keys.map(key => optionFormat(key));
+
+  const handleBlur = () => onBlur({ target: { name, value } });
+
+  const handleChangeKey = (selected, name) => {
+    if (selected === null) {
+      onChange({ target: { name, value: '' } });
+    } else {
+      const { value } = selected;
+      onChange({ target: { name, value } });
+    }
+  };
+
+  const handleClick = () => onClick(name);
+
+  const handleRemoveItem = index => {
+    const event = index === 0 && value.length === 1 ? 'clear' : 'remove';
+    onRemove({ event, index });
+  };
+
+  const customStyles = hasError => {
+    const selectBorder = isFocused => {
+      if (isFocused) {
+        return '1px solid #78caff !important';
+      }
+      if (hasError) {
+        return '1px solid #F64D0A !important';
+      }
+      return '1px solid #E3E9F3 !important';
+    };
+
+    return {
+      control: (base, state) => ({
+        ...base,
+        border: selectBorder(state.isFocused),
+        borderRadius: '2px !important',
+      }),
+      menu: base => {
+        return {
+          ...base,
+          padding: '0',
+          border: '1px solid #e3e9f3',
+          borderTop: '1px solid #78caff',
+          borderTopRightRadius: '0',
+          borderTopLeftRadius: '0',
+          borderBottomRightRadius: '3px',
+          borderBottomLeftRadius: '3px',
+          boxShadow: 'none',
+          marginTop: '-1px;',
+        };
+      },
+      menuList: base => ({
+        ...base,
+        maxHeight: '224px',
+        paddingTop: '0',
+      }),
+      option: (base, state) => {
+        return {
+          ...base,
+          backgroundColor:
+            state.isSelected || state.isFocused ? '#f6f6f6' : '#fff',
+          color: '#000000',
+          fontSize: '13px',
+          fontWeight: state.isSelected ? '600' : '400',
+          cursor: state.isFocused ? 'pointer' : 'initial',
+          height: '32px',
+          lineHeight: '16px',
+        };
+      },
+    };
+  };
+
+  return (
+    <Wrapper>
+      <ul>
+        <li>
+          <section>
+            <p>
+              <FormattedMessage id="Settings.webhooks.key" />
+            </p>
+          </section>
+          <section>
+            <p>
+              <FormattedMessage id="Settings.webhooks.value" />
+            </p>
+          </section>
+        </li>
+        {value.map((header, index) => {
+          const { key, value } = header;
+          const entryErrors = get(errors, index, null);
```

Comment:
You will achieve the same result for getting the error if you use directly the errors from the `getYupInnerErrors` utils. 
(c.f. feedback below)

Reply:
As I said below, I thought it was cleaner to pass a hierarchical object instead of an objects with concatenated keys. 

Do you think it's better that the HeaderInput itself manage its own errors with an object where keys are path?

Reply:
In fact this is how it is done in both the content manager and the content type builder.. until we develop hooks to improve the way we code forms I strongly believe that we should apply the same logic in all the administration panel so yes I think it's better that the header input "manages" its own errors which is basically retrieving the corresponding key from the error object

Reply:
Ok i see, i will adapt it!